source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.h

```diff
@@ -12,6 +12,19 @@ namespace Network {
 
 DECLARE_FACTORY(GetAddrInfoDnsResolverFactory);
 
+// Trace information for getaddrinfo.
+enum class GetAddrInfoTrace : uint8_t {
+  NotStarted = 0,
```

Comment:
I would have probably gone with a letter per state since that tends to be more readable, but it's also more computationally complex so I'm fine either way.  cc @abeyad

Reply:
what do you mean by a letter per state?  like `A`, `B`, etc?